src/internal/Schemable2.ts

```diff
@@ -0,0 +1,117 @@
+import { HKT2, Kind2, URIS2 } from 'fp-ts/HKT'
+import { Literal } from 'io-ts/Schemable'
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Any = any
+
+type AofHKT2<S, X> = [X] extends [HKT2<S, Any, infer A>] ? A : never
+type EofHKT2<S, X> = [X] extends [HKT2<S, infer E, Any>] ? E : never
+
+type AofKind2<S extends URIS2, X> = [X] extends [Kind2<S, Any, infer A>] ? A : never
+type EofKind2<S extends URIS2, X> = [X] extends [Kind2<S, infer E, Any>] ? E : never
+
+type EnsureTagHKT2<S, T extends string, M> = {
+  [K in keyof M]: HKT2<S, Any, { [_ in T]: K }>
+}
+
+type EnsureTagKind2<S extends URIS2, T extends string, M> = {
+  [K in keyof M]: Kind2<S, Any, { [_ in T]: K }>
+}
+
+export interface SchemableHKT2<S> {
```

Comment:
The `SchemaExt` function must return an `HKT2` in order to properly derive an Encoder, otherwise there would be no type on the `E` parameter. I've basically replaced the `HKT` one with the `HKT2` one in this PR by having `SchemableExt` extend `SchemableHKT2` instead of the `Schemable` from `io-ts`

https://github.com/jacob-alford/schemable-ts-types/blob/40481e15bbd139aef6bff7c13a1a8480a04bd317/src/SchemableExt.ts#L46